docs/data-retention-mechanisms.md

```diff
@@ -0,0 +1,65 @@
+# Data retention mechanisms
+
+There are several mechanisms regarding the retention of data in EmailAlertApi.
+This doc gives a brief overview of each.
+
+## Deactivating subscribers
+
+We deactivate subscribers for a couple of different scenarios. One being where
+the user unsubscribes via either following an unsubscribe link in the email
+or via the manage subscriptions page.
+
+Another way a user is deactivated is if their subscription is ended on our side
+because the topic they're signed up for no longer exists. This happens via an
+unpublished rabbitMQ message which is eventually handled by the
+[UnpublishHandleService]. This is only applicable for subscriptions to the
+topic taxonomy.
+
+[UnpublishHandleService]: https://github.com/alphagov/email-alert-api/blob/master/app/services/unpublish_handler_service.rb
+
+## Nullifying deactivated subscribers
+
+We nullify the email addresses of subscribers after they've been deactivated
+for over 28 days. This means that the address field on the subscriber record is
+updated to nil once the threshold has been reached.
+
+We use the [NullifyDeactivatedSubscribersWorker] to do this which runs [every
+hour].
+
+[every hour]: https://github.com/alphagov/email-alert-api/blob/master/config/sidekiq.yml#L26-L28
+[NullifyDeactivatedSubscribersWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/nullify_deactivated_subscribers_worker.rb
+
+## Archiving emails
+
+We archive emails to S3 using the [EmailArchiveWorker]. Emails are eligible for
+archiving immediately after creation as long as they haven't already been
+archived (based on `archived_at` value).
+
+The [archive worker runs every hour].
+
+[EmailArchiveWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[archive worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L29-L31
+
+## Deleting emails
+
+We delete emails after they've been archived for a minimum of 7 days. Emails get
+archived immediately (as soon as they're generated) and then deleted using the
+[EmailDeletionWorker].
+
+The [deletion worker runs every hour].
+
+[deletion worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L32-L34
+[EmailDeletionWorker]: https://github.com/alphagov/email-alert-api/blob/a62abc85453b723d683c2dc13f3bf0065fb86d5f/app/workers/email_deletion_worker.rb
+
+## Deleting historic data
+
+We now delete data we deem as historic everday at [midday] using the
+[HistoricalDataDeletionWorker].
+
+The data we delete is deemed historic if its now redundant and has no use
```

Comment:
We can be clearer here: we're _asserting_ that data over a year old has no use in the system, or analytical purposes. I think it's also important to mention the motivations: it makes analytics harder; and some of it quickly fills the DB.

Reply:
I've added a bit more detail and the motivations which improves the section. 

I think we need to be careful though using 'over a year' as that implies an out and out rule but thats not the case for some cases.